File: _examples/cpp/tuples.cpp
#include <tuple>

// 创建元组
std::tuple<int, std::string, double> person(25, "Alice", 1.68);

// 访问元素
int age = std::get<0>(person);
std::string name = std::get<1>(person);

// C++17 结构化绑定
auto [a, n, h] = person;

File: _examples/zig/tuples.zig
const std = @import("std");

// Zig 使用结构体模拟元组
const Tuple = struct { i32, []const u8, f64 };

const person = Tuple{ 25, "Alice", 1.68 };
const age = person[0];
const name = person[1];

// 单元素元组
const single = struct { i32 }{ 42 };
